core/src/main/java/org/apache/carbondata/core/datastore/filesystem/AbstractDFSCarbonFile.java

```diff
@@ -331,6 +378,26 @@ public boolean delete() {
     return fs.create(pt, true, bufferSize, fs.getDefaultReplication(pt), blockSize);
   }
 
+  @Override public DataOutputStream getDataOutputStream(String path, FileFactory.FileType fileType,
```

Comment:
It already has a method in this class , change that method to use codec as per the file extension